[BACKLOG-5838] Prompt API - Implemented showProgressIndicator

annapopova1 opened the pull request:
Sample framework: https://github.com/pentaho/pentaho-platform-ee/pull/860
@diogofscmariano @plagoa @krivera-pentaho @pamval  @afrjorge  please review

krivera-pentaho commented on package-res/resources/web/prompting/api/UiAPI.js:
```diff
@@ -15,8 +15,25 @@
   return function(api) {
-
+    /**
+     * Makes visible the progress indicator by calling function Dashboard#showProgressIndicator.
```
I would change this line to "Makes the progress indicator visible by calling Dashboard#showProgressIndicator"

krivera-pentaho commented on package-res/resources/web/prompting/api/UiAPI.js:
```diff
@@ -15,8 +15,25 @@
 
+/**
+ * The Prompting UI API Class
+ * Provides general functions for working with UI of prompting.
```
Change to "Provides general functions for working with the UI of prompting."

annapopova1 commented:
@krivera-pentaho 
Virginia suggested another sentences for this API:
for class:
"This is the prompting user interface API class. Contains all general functions for working with the prompting user interface."

for method:
"Shows the progress indicator by calling the function Dashboard#showProgressIndicator."

Do you agree?

pamval commented on package-res/resources/web/prompting/api/UiAPI.js:
```diff
@@ -15,8 +15,25 @@
+     *     api.ui.showProgressIndicator();
+     */
+    this.showProgressIndicator = function() {
+      api.operation._getPromptPanel().dashboard.showProgressIndicator();
```
I'd probably avoid dashboard references to creep up to this level of the API.
This code assumes that a prompt panel will always contain a dashboard object that implements a showProgressIndicator method.

If we are targeting being able to change the prompt renderer from CDF to something else, that would mean that we'd have to change this function to achieve that, which feels wrong.

The easy fix would be for promptPanel to encapsulate how it shows/hides the progress indicator, therefore making the dashboard reference internal to it.

krivera-pentaho replied:
I completely agree with this.

krivera-pentaho commented:
Yes Anna, I agree with the suggested changes.

annapopova1 commented:
@pamval @krivera-pentaho could you take a look?
- changed docs as Virginia proposed;
- added showProgressIndicator function to PromptPanel to encapsulate dashboard's logic.

annapopova1 commented:
changed docs as Virginia proposed one more time

diogofscmariano commented:
@annapopova1 can we have the documentation extended with what we described in the meeting? Something like when this is called, what this does

diogofscmariano commented:
@annapopova1 do we have these functions used in the prompt panel? does it makes sense to make sure within the prompt panel only these are being used and not others?

annapopova1 commented:
@diogofscmariano extended documentation.
This functions are not used in PromptPanel. I put it to PromptPanel for encapsulate operations with dashboard object (depends of CDF implementation). Pedro wrote the comment about it above.

annapopova1 commented:
@diogofscmariano  Implemented OperationAPI#refreshPrompt

diogofscmariano commented:
@annapopova1 can't merge it. please fix the conflicts

annapopova1 commented:
@diogofscmariano fixed conflicts
